### ISSUE TITLE
CLI az vm update to support Dedicated Host ID assignment

### ISSUE THREAD

**Issue by brittanyrowe** [Azure/azure-cli#19622]
**Is your feature request related to a problem? Please describe.**
Azure Dedicated Host (ADH) customers can already assign an existing VM to a specific ADH manually via Portal or by specifying the host ID through PS, but not on CLI. This is a problem for customers who would like to use CLI to manage their dedicated hosts and the VMs on them. We would like to have CLI and PS parity.

**Describe the solution you'd like**
 Customers can call the **az vm update** command on CLI with a specified Dedicated Host ID to assign the VM to that host. 

**Describe alternatives you've considered**
Routing customers through PS is not always an option, especially if they are Linux users. Going through the Portal is another alternative. 

**Additional context**

- ADH documentation for assigning a VM on PS: https://docs.microsoft.com/en-us/azure/virtual-machines/dedicated-hosts-how-to?tabs=portal%2Cpowershell2#add-an-existing-vm
- az vm create already supports the host ID (--host) parameter: https://docs.microsoft.com/en-us/cli/azure/vm?view=azure-cli-latest#az_vm_create

**Comment by yonzhan** [comment]
Compute

**Comment by wangzelin007** [comment]
@brittanyrowe Hi, I want to confirm do you also want add Dedicated Host Group --host-group argument this time ?

**Comment by brittanyrowe** [comment]
@wangzelin007 yes, let's also include --host-group so autoplaced VMs can be assigned to a dedicated host group.

**Comment by wangzelin007** [comment]
> @wangzelin007 yes, let's also include --host-group so autoplaced VMs can be assigned to a dedicated host group.

@brittanyrowe Hi, I want to confirm again, I found that the following three situations are not supported during the test:
1. If the host and host-group are not specified when create a vm, the specified host and host-group will not take effect when update the vm.
2. If a host is specified when create a vm, the specified host-group  will not take effect when update the vm.
3. If a host-group is specified when create a vm, the specified host will not take effect when update the vm.
